lib/errors.ts

```diff
@@ -0,0 +1,254 @@
+/*
+Copyright 2016 Resin.io
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/// <reference path='ext.d.ts'/>
+
+/**
+ * @module errors
+ */
+
+import TypedError = require('typed-error');
+
+export class ResinError extends TypedError {
+	public code: string;
+	public exitCode: number;
+}
+ResinError.prototype.code = 'ResinError';
+ResinError.prototype.exitCode = 1;
```

Comment:
I did not realise that, but you're totally right: https://github.com/Microsoft/TypeScript/issues/1250

Fair enough, in that case this is fine by me :+1: